presto-prometheus/src/main/java/com.facebook.presto.plugin.prometheus/PrometheusRecordCursor.java

```diff
@@ -0,0 +1,298 @@
+            fieldToColumnIndex[i] = columnHandle.getOrdinalPosition();
+        }
+
+        try (CountingInputStream input = new CountingInputStream(byteSource.openStream())) {
+            metricsItr = prometheusResultsInStandardizedForm(new PrometheusQueryResponseParse(input).getResults()).iterator();
+            totalBytes = input.getCount();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    static Block getBlockFromMap(Type mapType, Map<?, ?> map)
+    {
+        // on functions like COUNT() the Type won't be a MapType
+        if (!(mapType instanceof MapType)) {
+            return null;
+        }
+        Type keyType = mapType.getTypeParameters().get(0);
+        Type valueType = mapType.getTypeParameters().get(1);
+
+        BlockBuilder mapBlockBuilder = mapType.createBlockBuilder(null, 1);
+        BlockBuilder builder = mapBlockBuilder.beginBlockEntry();
+
+        for (Map.Entry<?, ?> entry : map.entrySet()) {
+            writeObject(builder, keyType, entry.getKey());
+            writeObject(builder, valueType, entry.getValue());
+        }
+
+        mapBlockBuilder.closeEntry();
+        return (Block) mapType.getObject(mapBlockBuilder, 0);
+    }
+
+    static Map<Object, Object> getMapFromBlock(Type type, Block block)
+    {
+        MapType mapType = (MapType) type;
+        Type keyType = mapType.getKeyType();
+        Type valueType = mapType.getValueType();
+        Map<Object, Object> map = new HashMap<>(block.getPositionCount() / 2);
+        for (int i = 0; i < block.getPositionCount(); i += 2) {
+            map.put(readObject(keyType, block, i), readObject(valueType, block, i + 1));
+        }
+        return map;
+    }
+
+    private static void writeObject(BlockBuilder builder, Type type, Object obj)
+    {
+        if (type instanceof ArrayType) {
+            Type elementType = ((ArrayType) type).getElementType();
+            BlockBuilder arrayBuilder = builder.beginBlockEntry();
+            for (Object item : (List<?>) obj) {
+                writeObject(arrayBuilder, elementType, item);
+            }
+            builder.closeEntry();
+        }
+        else if (type instanceof MapType) {
+            MapType mapType = (MapType) type;
+            BlockBuilder mapBlockBuilder = builder.beginBlockEntry();
+            for (Map.Entry<?, ?> entry : ((Map<?, ?>) obj).entrySet()) {
+                writeObject(mapBlockBuilder, mapType.getKeyType(), entry.getKey());
+                writeObject(mapBlockBuilder, mapType.getValueType(), entry.getValue());
+            }
+            builder.closeEntry();
+        }
+        else {
+            if (BOOLEAN.equals(type)
+                    || TINYINT.equals(type)
+                    || SMALLINT.equals(type)
+                    || INTEGER.equals(type)
+                    || BIGINT.equals(type)
+                    || DOUBLE.equals(type)
```

Comment:
This is always going to be false as we are comparing `String` with `Type`

Reply:
I'm not sure I understand what you mean here. Type is an interface but looks like the caller will pass in from getKeyType or getTypeParameters. I verify using TestPrometheusMetricsIntegration and break at the line to check which evaluates correctly.